src/python/pants/engine/internals/specs_rules.py

```diff
@@ -207,7 +208,13 @@ async def addresses_from_raw_specs_with_only_file_owners(
     all_files = tuple(itertools.chain.from_iterable(paths.files for paths in paths_per_include))
     owners = await Get(
         Owners,
-        OwnersRequest(all_files, filter_by_global_options=specs.filter_by_global_options),
+        OwnersRequest(
+            all_files,
+            filter_by_global_options=specs.filter_by_global_options,
+            match_if_owning_build_file_included_in_sources=(
+                cli_subsystem.build_files_expand_to_targets
+            ),
```

Comment:
This will trigger the deprecation warning for any command that uses a file literal, right?

It would be really good to avoid these "everyone sees it" type deprecations where possible until/unless we support generating a `pants.toml` for folks that silences all of the deprecations for new users. When it's unconditional like this, the audience for the deprecation ends up being the administrator (because they'll toggle the setting and then move on with their lives): it doesn't actually reach the end user, who is the actual consumer who needs to change their behavior.

In this case, that might mean only triggering the deprecation if we actually encounter a `BUILD` file as a CLI argument.

Reply:
Yeah, this is currently an unconditional deprecation, which indeed is a real bummer for new users. Hence part of my desire to rip out the "buggy" behavior w/o deprecation.

> In this case, that might mean only triggering the deprecation if we actually encounter a BUILD file as a CLI argument.

Unfortunately that will happen pretty quickly thanks to globs like `./pants fmt ::`.

Reply:
> > In this case, that might mean only triggering the deprecation if we actually encounter a BUILD file as a CLI argument.
> 
> Unfortunately that will happen pretty quickly thanks to globs like `./pants fmt ::`.

It seems like it should be possible to move the deprecation then? Perhaps to "use of a file literal that matches the BUILD patterns"?

Reply:
I think that could work, I only need to verify the results of globs aren't impacted at all by this option. I'll try to verify that tomorrow. Thanks!

Reply:
I do think that "treat the BUILD file path as if it were all the targets in that BUILD file" is not intended or expected behavior, and we can treat it as a bug.

Reply:
I changed this to have no option and be a "bug fix" -- PTAL